lib/util/http/auth.ts

```diff
@@ -6,7 +6,20 @@ import {
 } from '../../constants';
 import type { GotOptions } from './types';
 
-export function applyAuthorization(inOptions: GotOptions): GotOptions {
+type AuthGotOptions = Pick<
```

Comment:
what's the benefit of doing this additional type?

Reply:
Otherwise it will lose `method` strictness